lib/octopus_serverspec_extensions.rb

```diff
@@ -15,3 +16,69 @@
 require 'octopus_serverspec_extensions/matcher/have_windows_line_endings.rb'
 require 'octopus_serverspec_extensions/matcher/have_linux_line_endings.rb'
 require 'octopus_serverspec_extensions/version.rb'
+
+# shared
+
+private
+
+def get_env_var(name)
+  raise 'non-approved env var' if name != 'OCTOPUS_CLI_API_KEY' && name != 'OCTOPUS_CLI_SERVER'
+  raise "env var #{name} not found" if ENV[name].nil?
+  ENV[name]
+end
+
+def get_octopus_url(server_url)
+  # returns the url or nil
+  if server_url.nil?
+    server_url = get_env_var('OCTOPUS_CLI_SERVER')
+  end
+
+  server_url
+end
+
+def get_octopus_api_key(api_key)
+  # returns the api key or nil
+  if api_key.nil?
+    api_key = get_env_var('OCTOPUS_CLI_API_KEY')
+  end
+
+  api_key
+end
+
+def get_octopus_creds(args)
+  server = args[0]
+  api_key = args[1]
+
+  if args.length != 0 && args.length != 2
+    raise "Supplied credentials invalid. Expected: [url, api_key] Received: #{args}"
+  end
+
+  if server.nil?
+    server = get_env_var('OCTOPUS_CLI_SERVER')
+  end
+
+  if api_key.nil?
+    api_key = get_env_var('OCTOPUS_CLI_API_KEY')
+  end
+
+  # are they still nil? raise an error
+  if api_key.nil? or server.nil?
+    raise "Supplied credentials invalid. One or more of [server, api_key] was null. " +
+              "If you intended to use Environment Variables, please check the value of OCTOPUS_CLI_SERVER and OCTOPUS_CLI_API_KEY"
+  end
+
+  server = server.chomp("/") # remove the trailing slash if it exists
+
+  [server, api_key]
+end
+
+def check_supports_spaces(server_url)
+  begin
+    resp = Net::HTTP.get_response(URI.parse("#{server_url}/api/"))
+    body = JSON.parse(resp.body)
+    version = body['Version']
+    return Gem::Version.new(version) > Gem::Version.new('2019.0.0')
```

Comment:
I _wonder_... should we be feature detecting rather than version detecting?
What if we decided to change back to 5.x?

Reply:
It's definitely a thought. At least we have shared code here so we'd only have to change it in one place.

Reply:
What would we do, detect if the `spaces` is mentioned under `links` on the `/api` root endpoint maybe? I can card that and look at it once we've smashed through these changes

Reply:
carded